tests/main/debug-migrate-home/task.yaml

```diff
@@ -0,0 +1,38 @@
+summary: Check the debug command migrate-home migrates snaps' homes
+
+prepare: |
+  "$TESTSTOOLS"/snaps-state install-local "test-snapd-tools"
+  "$TESTSTOOLS"/snaps-state install-local basic
+
+restore: |
+  snap unset system experimental.move-snap-home-dir
+  snap remove --purge basic
+  snap remove --purge test-snapd-tools
+
+execute: |
+  echo "Check that migrate home doesn't work without setting the experimental flag"
+  if got=$(snap debug migrate-home test-snapd-tools 2>&1); then
+    echo 'Calling "snap debug migrate-home" without setting "experimental.move-snap-home-dir" should fail'
+    exit 1
+  fi
+  #shellcheck disable=SC2086
+  echo $got |  MATCH 'error: cannot migrate to ~/Snap: flag "experimental.move-snap-home-dir" is not set'
+
+  echo "Check that migrate home migrates the data under SNAP_USER_DATA"
+  snap set system experimental.move-snap-home-dir=true
+
+  #shellcheck disable=SC2016
+  test-snapd-tools.cmd sh -c 'echo foo > $HOME/foo'
+  MATCH "foo" < "$HOME/snap/test-snapd-tools/current/foo"
+  not test -e "$HOME"/Snap
+
+  snap debug migrate-home test-snapd-tools basic
+
+  MATCH "foo" < "$HOME/Snap/test-snapd-tools/foo"
```

Comment:
Should we also have
```
not test -e "$HOME/snap/test-snapd-tools/current/foo"
```
?

Reply:
This migration doesn't replace that directory with the one under ~/Snap. The point is just to create a directory under ~/Snap (which isn't revisioned) to serve as the new home directory and to initialize it with the contents of the previous home (which was also SNAP_USER_DATA). The revisioned directory remains in place and keeps serving as a place for snap data storage